docusaurus/docs/cloud/advanced/upload.md

```diff
@@ -0,0 +1,204 @@
+        baseUrl: env('CDN_URL'),
+        rootPath: env('CDN_ROOT_PATH'),
+        s3Options: {
+          accessKeyId: env('AWS_ACCESS_KEY_ID'),
+          secretAccessKey: env('AWS_ACCESS_SECRET'),
+          region: env('AWS_REGION'),
+          params: {
+            ACL: env('AWS_ACL', 'public-read'),
+            signedUrlExpires: env('AWS_SIGNED_URL_EXPIRES', 15 * 60),
+            Bucket: env('AWS_BUCKET'),
+          },
+        },
+      },
+      actionOptions: {
+        upload: {},
+        uploadStream: {},
+        delete: {},
+      },
+    },
+  },
+  // ...
+});
+```
+</TabItem>
+</Tabs>
+
+## Security Middleware Configuration
+Due to the default settings in the Strapi Security Middleware you will need to modify the `contentSecurityPolicy` settings to properly see thumbnail previews in the Media Library.
+
+To do this in your Strapi project:
+1. Navigate to `./config/middleware.js` or `./config/middleware.ts` in your Strapi project.
+2. Replace the default `strapi::security` string with the object provided by the upload provider.
+
+Examples of configuring the `contentSecurityPolicy` depending on upload provider:
+
+<Tabs groupId="upload-examples" >
+<TabItem value="cloudinary" label="Cloudinary">
+
+```js
+module.exports = [
+  // ...
+  {
+    name: 'strapi::security',
+    config: {
+      contentSecurityPolicy: {
+        useDefaults: true,
+        directives: {
+          'connect-src': ["'self'", 'https:'],
+          'img-src': ["'self'", 'data:', 'blob:', 'market-assets.strapi.io', 'res.cloudinary.com'],
+          'media-src': [
+            "'self'",
+            'data:',
+            'blob:',
+            'market-assets.strapi.io',
+            'res.cloudinary.com',
+          ],
+          upgradeInsecureRequests: null,
+        },
+      },
+    },
+  },
+  // ...
+];
+```
+</TabItem>
+<TabItem value="amazon-s3" label="Amazon S3">
+
+```js
+module.exports = [
+  // ...
+  {
+    name: 'strapi::security',
+    config: {
+      contentSecurityPolicy: {
+        useDefaults: true,
+        directives: {
+          'connect-src': ["'self'", 'https:'],
+          'img-src': [
+            "'self'",
+            'data:',
+            'blob:',
+            'market-assets.strapi.io',
+            'yourBucketName.s3.yourRegion.amazonaws.com',
+          ],
+          'media-src': [
+            "'self'",
+            'data:',
+            'blob:',
+            'market-assets.strapi.io',
+            'yourBucketName.s3.yourRegion.amazonaws.com',
+          ],
+          upgradeInsecureRequests: null,
+        },
+      },
+    },
+  },
+  // ...
+];
+```
+</TabItem>
+</Tabs>
+
+
+
+
+
+:::caution
+The upload provider must be installed as a package dependency in your Strapi project. For example, to use the Cloudinary upload provider, install the `@strapi/provider-upload-cloudinary` package.
+
+For specific configuration information, see the [Marketplace](https://market.strapi.io/) entry for that provider.
+:::
```

Comment:
Same feedback as in "Email Provider", we should write the steps in the right order 😊